app.rb

```diff
@@ -0,0 +1,151 @@
+# Import necessary classes
+require_relative 'book'
+require_relative 'classroom'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+# Define the main application class
+class App
+  # Create getters and setters for books, people, and rentals
+  attr_accessor :books, :people, :rentals
+
+  # Initialize the App class with empty arrays for books, people, and rentals
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  # List all books in the books array
+  def list_books
+    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  # List all books with their indices
+  def list_books_with_index
+    @books.each_with_index { |book, i| puts "#{i}) Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  # List all people in the people array
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  # List all people with their indices
+  def list_people_with_index
+    @people.each_with_index do |person, i|
+      puts "#{i}) [#{person.class.name}] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  # Create a person (either a student or a teacher) based on user input
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [input the number]: '
+    student_or_teacher = gets.chomp.to_i
+    case student_or_teacher
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts "Invalid choice. Please enter a valid option. (#{student_or_teacher})"
+    end
+  end
+
+  # Create a student based on user input
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp.to_s
+
+    print 'Has parent permission? [Y / N]: '
+    parent_permission = gets.chomp.to_s
+
+    if parent_permission =~ /^[Yy]/
+      student = Student.new('Unknown', age, name, parent_permission: true)
+    elsif parent_permission =~ /^[Nn]/
+      student = Student.new('Unknown', age, name, parent_permission: false)
+    else
+      puts "Invalid choice. Please enter a valid option. (#{parent_permission})"
+      return
+    end
```

Comment:
- [ ] When I try to create a `Student` instance and a Teacher Instance, I get these errors:

![image](https://github.com/HossainAraf/school-library/assets/77106962/5e91e191-e96a-4d06-ab42-417c84adfe77)

![image](https://github.com/HossainAraf/school-library/assets/77106962/c09ab33e-c9b1-44e0-a94b-8b7ccf73a711)

---
During the initialization of the `Person` class, you made the `parent_permission` parameter a keyword parameter. Therefore when calling the `super` method in the `initialize` method of these `Teacher` and `Student` classes, you need to specify the value of the parent_permission parameter. I would advise you to change the initialize method of the classes to look like this 👍 💻:

```ruby
# For Student class

def initialize(age, classroom, name = 'unknown', parent_permission: true)
  super(age, name, parent_permission: parent_permission)
  @classroom = classroom
 end
 
 # For Teacher class

def initialize(specialization, age, name = 'Unknown', parent_permission: true)
  super(age, name, parent_permission: parent_permission)
  @specialization = specialization
 end
```